modules/lightmapper_rd/lightmapper_rd.cpp

```diff
@@ -1043,10 +1046,16 @@ LightmapperRD::BakeError LightmapperRD::bake(BakeQuality p_quality, bool p_use_d
 	RenderingDevice *rd = RenderingServer::get_singleton()->create_local_rendering_device();
 	if (rd == nullptr) {
 #if defined(RD_ENABLED)
-#if defined(VULKAN_ENABLED)
-		rcd = memnew(RenderingContextDriverVulkan);
+#if defined(METAL_ENABLED)
+		rcd = memnew(RenderingContextDriverMetal);
 		rd = memnew(RenderingDevice);
 #endif
+#if defined(VULKAN_ENABLED)
+		if (rcd == nullptr) {
+			rcd = memnew(RenderingContextDriverVulkan);
+			rd = memnew(RenderingDevice);
+		}
+#endif
```

Comment:
I'm not too familiar with local RenderingDevice, but shouldn't we try to follow the project's configured rendering method instead of trying them in arbitrary order?

I see here it's Metal first, then Vulkan, but DisplayServer does Vulkan -> D3D12 -> Metal. Is it ok to mix and match though, assuming several of them work? Or will the successive `memnew(RenderingContextDriver*)` fail until the currently used rendering driver is tried?

Reply:
In the case of display server, I believe that is just determining if an RD device is available, so order is arbitrary.

Here, the order is to prefer the platforms native API, so Metal first. If D3212 is added that would also appear before `VULKAN_ENABLED`.

Reply:
In this case, the default behaviour is to use the projects preferred device for betsy and lightmapper. This code is only when using the compatibility renderer, according to @BlueCube3310, so we choose the platforms preferred graphics API.

Reply:
> shouldn't we try to follow the project's configured rendering method instead of trying them in arbitrary order?

I may be wrong, but I think this whole section is for when the project _isn't_ configured to use RD, because we use RD for baking lightmaps even when the project itself uses OpenGL.

If the project _is_ using RD, then it will use what the project has configured, which is the case this line a little bit back is for: https://github.com/godotengine/godot/pull/96351/files#diff-bcf125f38354f39b72231f6bbe399962aedb85101657d1a00f3a002b4f81891fL1043